pkg/controller/command/mediator/command.go

```diff
@@ -244,6 +269,74 @@ func (o *Command) Reconnect(rw io.Writer, req io.Reader) command.Error {
 	return nil
 }
 
+// ReconnectAll performs reconnection on all available mediator connections.
+// This command is useful in re-establishing lost connections (ex: lost websocket connection).
+func (o *Command) ReconnectAll(rw io.Writer, req io.Reader) command.Error {
+	connections, err := o.routeClient.GetConnections()
+	if err != nil {
+		logutil.LogError(logger, CommandName, ReconnectAllCommandMethod, err.Error())
+
+		return command.NewExecuteError(ReconnectAllError, err)
+	}
+
+	for _, connection := range connections {
+		err = o.messageClient.Noop(connection)
+		if err != nil {
+			logutil.LogError(logger, CommandName, ReconnectAllCommandMethod, err.Error(),
+				logutil.CreateKeyValueString(connectionID, connection))
+			return command.NewExecuteError(ReconnectRouterErrorCode, err)
+		}
+	}
+
+	command.WriteNillableResponse(rw, nil, logger)
+
+	logutil.LogDebug(logger, CommandName, ReconnectAllCommandMethod, successString)
+
+	return nil
+}
+
+// CreateInvitation creates out-of-band invitation from one of the mediator connections.
+func (o *Command) CreateInvitation(rw io.Writer, req io.Reader) command.Error {
+	connections, err := o.routeClient.GetConnections()
+	if err != nil {
```

Comment:
I see that this function only adds a random connection. I don't think that we need a new function for that.
I would suggest modifying the existing function (outOfBand.CreateInvitation) to support random `router connection`. 
To be honest, I don't like this random pick up at all. I think it should be part of edge-agent refactoring. After we set up the router we have a connection, so we can use it everywhere we need it.

@sudeshrshetty @llorllale @rolsonquadras ^

Reply:
Nothing wrong in adding a command to reduce number of lines or reduce repeated logic for clients. This is moved from agent-sdk.

if you pass all connections then out of band client picks first one always, which is not what we need for router.

As comment says, this command is for clients who can directly create invitation without passing any connection details, like `mediator.createInvitation()`.  Mediator will pickup one random connection (current webwallet logic).

If someone wants to create invitation for particular connection, he doesn't need this command.

Reply:
I agree the randomization of the router selection should be exposed in the clients and not implemented here.

Reply:
We should not be reimplementing this logic everywhere.

Reply:
removed `mediator.createInvitation()`